gcloud/storage/bucket.py

```diff
@@ -222,14 +227,22 @@ def get_blob(self, blob_name, client=None):
         :param client: Optional. The client to use.  If not passed, falls back
                        to the ``client`` stored on the current bucket.
 
+        :type generation: int
+        :param generation: Optional. The generation id to retrieve in a bucket
+                           that supports versioning.
+
         :rtype: :class:`gcloud.storage.blob.Blob` or None
         :returns: The blob object if it exists, otherwise None.
         """
         client = self._require_client(client)
         blob = Blob(bucket=self, name=blob_name)
         try:
+            path = blob.path
+            if generation is not None:
+                path = (blob.path + '?generation=' +
+                        quote(str(generation), safe=''))
```

Comment:
I totally get why you'd go that route. A little of where how we envision these working together may help. We consider `Bucket.get_blob` and `Bucket.delete_blob` to be tiny do-nothing wrappers and put all verbs (i.e. API methods) relating to a noun (here the noun is `Blob`) on the relevant object.